models/brats_mri_generative_diffusion/docs/README.md

```diff
@@ -0,0 +1,165 @@
+
+## MONAI Generative Model Dependencies
+[MONAI generative models](https://github.com/Project-MONAI/GenerativeModels) can be installed by
+```
+pip install lpips==0.1.4
+pip install git+https://github.com/Project-MONAI/GenerativeModels.git@0.2.0
+```
+
+Or we can install it from source
+```
+pip install lpips==0.1.4
+git clone https://github.com/Project-MONAI/GenerativeModels.git
+cd GenerativeModels/
+python setup.py install
+cd ..
+```
+
+## Data
+The training data is BraTS 2016 and 2017 from [Medical Decathlon](http://medicaldecathlon.com/).
+
+- Target: Image Generation
+- Task: Synthesis
+- Modality: MRI
+- Size: 388 3D volumes (1 channel used)
+
+The dataset can be downloaded automatically at the beginning of training.
+
+## Training Configuration
+If you have a GPU with less than 32G of memory, you may need to decrease the batch size when training. To do so, modify the `train_batch_size` parameter in the [configs/train_autoencoder.json](configs/train_autoencoder.json) and [configs/train_diffusion.json](configs/train_diffusion.json) configuration files.
+
+### Training Configuration of Autoencoder
+The autoencoder was trained using the following configuration:
+
+- GPU: at least 32GB GPU memory
+- Actual Model Input: 112 x 128 x 80
+- AMP: False
+- Optimizer: Adam
+- Learning Rate: 1e-5
+- Loss: L1 loss, perceptual loss, KL divergence loss, adversarial loss, GAN BCE loss
+
+#### Input
+1 channel 3D MRI Flair patches
+
+#### Output
+- 1 channel 3D MRI reconstructed patches
+- 8 channel mean of latent features
+- 8 channel standard deviation of latent features
+
+### Training Configuration of Diffusion Model
+The latent diffusion model was trained using the following configuration:
+
+- GPU: at least 32GB GPU memory
+- Actual Model Input: 36 x 44 x 28
+- AMP: False
+- Optimizer: Adam
+- Learning Rate: 1e-5
+- Loss: MSE loss
+
+#### Training Input
+- 8 channel noisy latent features
+- an int that indicates the time step
+
+#### Training Output
+8 channel predicted added noise
+
+#### Inference Input
+8 channel noise
+
+#### Inference Output
+8 channel denoised latent features
+
+### Memory Consumption Warning
+
+If you face memory issues with data loading, you can lower the caching rate `cache_rate` in the configurations within range [0, 1] to minimize the System RAM requirements.
+
+## Performance
+
+#### Training Loss
+![A graph showing the autoencoder training curve](https://developer.download.nvidia.com/assets/Clara/Images/monai_brain_image_gen_ldm3d_train_autoencoder_loss.png)
+
+![A graph showing the latent diffusion training curve](https://developer.download.nvidia.com/assets/Clara/Images/monai_brain_image_gen_ldm3d_train_diffusion_loss.png)
+
+#### Example synthetic image
+![Example synthetic image](https://developer.download.nvidia.com/assets/Clara/Images/monai_brain_image_gen_ldm3d_example_generation.png)
+
+## MONAI Bundle Commands
```

Comment:
Updated to follow same style as 2D LDM

## MONAI Bundle Commands

In addition to the Pythonic APIs, a few command line interfaces (CLI) are provided to interact with the bundle. The CLI supports flexible use cases, such as overriding configs at runtime and predefining arguments in a file.

For more details usage instructions, visit the [MONAI Bundle Configuration Page](https://docs.monai.io/en/latest/config_syntax.html).

#### Execute Autoencoder Training (w/data download)

Assuming the current directory is the bundle directory, the following command will train the autoencoder network for 1500 epochs using the BraTS dataset. If the dataset is not downloaded, it will be automatically downloaded and extracted to `./Task01_BrainTumour`.

```
python -m monai.bundle run --config_file configs/train_autoencoder.json --dataset_dir ./ --download_brats True
```

#### Execute Autoencoder Training
If the dataset is already downloaded, make sure that `"dataset_dir"` in `configs/train_autoencoder.json` has the correct path to the dataset `Task01_BrainTumour`. Then, run:

```
python -m monai.bundle run --config_file configs/train_autoencoder.json
```

#### Override the `train` config to execute multi-GPU training for Autoencoder:
To train with multiple GPUs, use the following command, which requires scaling up the learning rate according to the number of GPUs.

```
torchrun --standalone --nnodes=1 --nproc_per_node=8 -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/multi_gpu_train_autoencoder.json']" --lr 8e-5
```

#### Execute Latent Diffusion Model Training
After training the autoencoder, run the following command to train the latent diffusion model. This command will print out the scale factor of the latent feature space. If your autoencoder is well trained, this value should be close to 1.0.

```
python -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/train_diffusion.json']"
```

#### Override the `train` config to execute multi-GPU training for Latent Diffusion Model:
To train with multiple GPUs, use the following command, which requires scaling up the learning rate according to the number of GPUs.

```
torchrun --standalone --nnodes=1 --nproc_per_node=8 -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/train_diffusion.json','configs/multi_gpu_train_autoencoder.json','configs/multi_gpu_train_diffusion.json']"  --lr 8e-5
```

#### Execute inference:
The following code generates a synthetic image from a random sampled noise.
```
python -m monai.bundle run --config_file configs/inference.json
```

#### Export checkpoint to TorchScript file:

The Autoencoder can be exported into a TorchScript file.

```
python -m monai.bundle ckpt_export autoencoder_def --filepath models/model_autoencoder.ts --ckpt_file models/model_autoencoder.pt --meta_file configs/metadata.json --config_file configs/inference.json
```